python/lib/physiological.py

```diff
@@ -549,6 +577,162 @@ def insert_event_file(self, event_data, event_file, physiological_file_id,
+        )
+
+        labels = []
+        for label, description in annotation_metadata['LabelDescription'].items():
+            descCount = self.db.pselect(
+                query="SELECT COUNT(*) FROM physiological_annotation_label WHERE LabelName = %s",
+                args=(label,)
+            )
+
+            if descCount == 0:
+                labels.insert((label, description))
+
+        self.db.insert(
+            table_name   = 'physiological_annotation_label',
+            column_names = ('LabelName', 'LabelDescription'),
+            values       = labels
+        )
+
+        # insert blake2b hash of annotation file into physiological_parameter_file
+        self.insert_physio_parameter_file(
+            physiological_file_id, 'annotation_file_blake2b_hash', blake2
+        )
+
+    def insert_annotation_data(self, annotation_data, annotation_file, physiological_file_id,
+                          blake2):
+        """
+        Inserts the annotation information read from the file *annotations.tsv
+        into the physiological_annotation_file and physiological_annotation_instance tables,
+        linking it to the physiological file ID already inserted in physiological_file.
+
+        :param annotation_data      : list with dictionaries of annotations
+                                      information to insert into
+                                      physiological_annotation_file
+         :type annotation_data      : list
+        :param annotation_file      : name of the annotation file
+         :type annotation_file      : str
+        :param physiological_file_id: PhysiologicalFileID to link the event info to
+         :type physiological_file_id: int
+        :param blake2               : blake2b hash of the task event file
+         :type blake2               : str
+        """
+
+        annotationFileID = self.db.insert(
+            table_name   = 'physiological_annotation_file',
+            column_names = ('PhysiologicalFileID', 'FileType',  'FilePath'),
+            values       = (physiological_file_id, 'tsv',  annotation_file),
+            get_last_id  = True
+        )
+
+        annotation_fields = (
+            'AnnotationFileID', 'AnnotationParameterID', 'Onset', 'Duration',
+            'AnnotationLabelID', 'Channel', 'AbsoluteTime', 'Description'
+        )
+        annotation_values = []
+        for row in annotation_data:
+            optional_fields = (
+                'channels', 'absolute_time', 'description'
+            )
+            for field in optional_fields:
+                if field not in row.keys():
+                    row[field] = None
+
+            if not row['duration'].isdecimal():
+                row['duration'] = None
+
+            if not row['onset'].isdecimal():
+                row['onset'] = None
+
+            if "n/a" in row['channels']:
+                row['channels'] = None
+
+            labelID = self.db.pselect(
```

Comment:
move this in a file `lib/database_lib/physiologicalannotationlabel.py` where all queries about physiological_annotation_label.py would go.